cmd/kustomize/docs/kustomize.yaml

```diff
@@ -1,15 +1,105 @@
-namePrefix: some-prefix
-# Labels to add to all objects and selectors.
-# These labels would also be used to form the selector for apply --prune
-# Named differently than “labels” to avoid confusion with metadata for this object
+# ----------------------------------------------------
+# Example kustomize.yaml content.
+#
+# This file declares the customization provided by
+# the kustomize program.
+#
+# Since customization is, by definition, _custom_,
+# there are no sensible default values for the fields
+# in this file.
+#
+# The field values used below are merely examples, not
+# to be copied literally.  The values won't work if
+# they happen to be references to external files that
+# don't exist.
+#
+# In practice, fields with no value should simply be
+# omitted from kustomize.yaml to reduce the content
+# visible in configuration reviews.
+# ----------------------------------------------------
+
+
+# Value of this field is prepended to the
+# names of all resources, e.g. a deployment named
+# "wordpress" becomes "alices-wordpress".
+namePrefix: alices-
+
+# Labels to add to all resources and selectors.
 labelsToAdd:
```

Comment:
`labelsToAdd` and `annotationsToAdd` violate API naming conventions. Too imperative.

https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#naming-conventions

`namePrefix` is not `stringToPrependToName`. :-)

commonLabels and commonAnnotations?

Reply:
I've used `additionalX` in the past, which is declarative (?) but I think feels imperative.

Reply:
Thanks for the feedback!

`commonLabels`, `additionalLabels` are nouns with adjectives.  `labelsToAdd` is a noun with a prepositional phrase functioning as an adjective.  `addLabels` would certainly be imperative.  

It would be nice to go with simply `labels` and `annotations`, to keep it consistent with (as Brian says) `namePrefix` and `resources`.

But shouldn't kustomization support _removal_?  If so, we want a struct that imposes no syntactic cost on people that have no need for removal (since we think removal would be rare).

Field names like `fooToAdd` and `fooToRemove` do this nicely - just omit `fooToRemove`.  Plus the names are self-documenting.

Examples of syntax that impose more cost on people with no need for removal are:
```
stuffToRemove {
  labels:
    fruit : apple
  resources:
  - map1.yaml
}
stuffToAdd {
  labels:  
    fruit : peach
  resources:
  - dep.yaml
  - map2.yaml
}
```
or the horrible
```
labels:
- operation: remove
  map:
    fruit: apple
- operation: add
  map:
    fruit: peach
resources:
- operation: remove
  - map1.yaml
- operation: add
  - dep.yaml
  - map2.yaml
}
```

Reply:
Simpler to not support delete, and say way.
 * don't add it to you base.
 * if you don't control the base, fork the base and remove it there.